feat(queue): add merge method fast-forward (#4534)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

   * - ``method``
     - string
     - ``merge``
     - Merge method to use. Possible values are ``merge``, ``squash``,
       ``rebase`` or ``fast-forward``.

>        ``fast-forward`` is not support on queue with ``speculative_checks > 1``,
>        ``batch_size > 1``, or with ``allow_inplace_checks`` set to ``false``

Reviewer: ```suggestion
       ``fast-forward`` is not support on queues with ``speculative_checks > 1``1,
       ``batch_size > 1``, or with ``allow_inplace_checks`` set to ``false``.
```

   * - ``update_method``
     - string

>      - ``merge`` for all merge method except ``fast-forward`` it's ``rebase``

Reviewer: ```suggestion
     - ``merge`` for all merge methods except ``fast-forward`` where ``rebase`` is used
```

       * ``merge`` to merge the base branch into the pull request.
       * ``rebase`` to rebase the pull request against its base branch.

>        This option is ignored if ``method`` is set to ``fast-forward``,
>        ``rebase`` will be always used.

Reviewer: ```suggestion
       This option is ignored if ``method`` is set to ``fast-forward``:
       ``rebase`` will be always used.
```

Can we raise rather than ignore? We could change the default to `None` where it'd pick the right method by default depending on the merge method.

Author: done
